syntax = "proto2";

package Arcusical.LocalMusicStore;

import "GUID.proto";

enum Encoding
{
	UNKNOWN_ENCODING = 0;
	AAC = 1;
	ALAC = 2;
	MP3 = 3;
	FLAC = 4;
	WAV = 5;
	ASF = 6;
}

enum Container
{
	UNKNOWN_CONTAINER = 0;
	MP3_CONTAINER = 1;
	MPEG4_CONTAINER = 2;
	FLAC_CONTAINER = 3;
	WAV_CONTAINER = 4;
	WMA_CONTAINER = 5;
}

message SongFile
{
	required Encoding encoding = 1;
	required string file = 2;
	required int32 bitRate = 3;
	required int32 sampleSize = 4;
	required int32 channelCount = 5;
	required Container container = 6;
}

message CachedSong
{
	required GUID id = 1;
	required string title = 2;
	required string artist = 4;
	required int64 length = 6;
	required string album = 7;
	repeated SongFile files = 8;
	required int32 trackNumber = 9;
	required int32 maxTrackNumber = 10;
	required int32 diskNumber = 11;
	required int32 maxDiskNumber = 12;
}